File: Python/Assignment4.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jun 29 19:49:02 2015

@author: Poonam
"""

import sys
import string
import logging

from Utils import mapper_output

logging.basicConfig(filename = mapper_output, format='%(message)s',
                    level = logging.INFO, filemode='w')

def mapper():    
    headers = []
    for line in sys.stdin:
        row = line.split(',')
        if not row[0]:
            headers = row
        else:
            row_with_header = filter(lambda x: (x[0] == "UNIT") or (x[0] == "ENTRIESn_hourly")
            , zip(headers, row))
            unit_entries = map(lambda x: x[1], row_with_header)
            logging.info("{0}\t{1}".format(unit_entries[0], unit_entries[1]))
        
mapper()

File: R/Assignment4.R
source("util.r")
library(logging)
basicConfig()
addHandler(writeToFile, file = mapper_output, level='INFO')

turnstile_weather <- read.table(input, header=T, sep="," )
head(turnstile_weather)

mapper <- function (turnstile_weather){
  Df <- turnstile_weather[,c("UNIT","ENTRIESn_hourly")]
  loginfo(paste(Df$UNIT, Df$ENTRIESn_hourly, sep = "\t"))
}

mapper(turnstile_weather)
